config/initializers/assets.rb

```diff
@@ -58,11 +58,12 @@ class << self
         # Manifests may be stored under the engine installation or under the Foreman app root, then
         # either with just the plugin name or with hyphens replaced with underscores.
         possible_manifests = [plugin.path, app.root].map do |root_dir|
+          next if root_dir.empty?
           [File.join(root_dir, "public/assets/#{plugin.id}/#{plugin.id}.json"),
            File.join(root_dir, "public/assets/#{plugin.id.to_s.tr('-', '_')}/#{plugin.id.to_s.tr('-', '_')}.json")]
         end.flatten
```

Comment:
You can use `compact` instead of `next if...`
```suggestion
        end.flatten.compact
```